volatility3/framework/plugins/linux/ptrace.py

```diff
@@ -0,0 +1,120 @@
+# This file is Copyright 2024 Volatility Foundation and licensed under the Volatility Software License 1.0
+# which is available at https://www.volatilityfoundation.org/license/vsl-v1.0
+#
+
+import logging
+from typing import List
+
+from volatility3.framework import renderers, interfaces, constants, objects
+from volatility3.framework.constants.linux import PT_FLAGS
+from volatility3.framework.constants.architectures import LINUX_ARCHS
+from volatility3.framework.objects import utility
+from volatility3.framework.configuration import requirements
+from volatility3.framework.interfaces import plugins
+from volatility3.plugins.linux import pslist
+
+vollog = logging.getLogger(__name__)
+
+
+class Ptrace(plugins.PluginInterface):
+    """Enumerates tracer and tracee tasks"""
+
+    _required_framework_version = (2, 10, 0)
+    _version = (1, 0, 0)
+
+    @classmethod
+    def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]:
+        return [
+            requirements.ModuleRequirement(
+                name="kernel",
+                description="Linux kernel",
+                architectures=LINUX_ARCHS,
+            ),
+            requirements.PluginRequirement(
+                name="pslist", plugin=pslist.PsList, version=(2, 2, 1)
+            ),
+        ]
+
+    @classmethod
+    def enumerate_ptraced_tasks(
+        cls,
+        context: interfaces.context.ContextInterface,
+        symbol_table: str,
+    ):
+        vmlinux = context.modules[symbol_table]
+
+        tsk_struct_symname = vmlinux.symbol_table_name + constants.BANG + "task_struct"
+
+        tasks = pslist.PsList.list_tasks(
+            context,
+            symbol_table,
+            filter_func=pslist.PsList.create_pid_filter(),
+            include_threads=True,
+        )
+
+        for task in tasks:
+            tracing_tid_list = [
+                int(task_being_traced.pid)
+                for task_being_traced in task.ptraced.to_list(
+                    tsk_struct_symname, "ptrace_entry"
+                )
+            ]
+
+            if task.ptrace == 0 and not tracing_tid_list:
+                continue
+
+            flags = (
+                PT_FLAGS(task.ptrace).flags
+                if task.ptrace != 0
+                else renderers.NotAvailableValue()
+            )
+
+            traced_by_tid = (
+                task.parent.pid
+                if task.real_parent != task.parent
+                else renderers.NotAvailableValue()
+            )
+
+            tracing_tids = ",".join(map(str, tracing_tid_list))
```

Comment:
This seems like a bad idea.  It's turning data into text form that then won't be possible to consume or work with if it's consumed by another plugin or a program.  I'd prefer this output separate lines with one entry per line please.  This might make the output quite verbose, I'm happy for you to use the Tree part of the TreeGrid to help organize these, but the point of having simple types returned is so that the data can be reused, not so it can be smushed into a human readable format and become more difficult for anything else to consume.